src/gui/mainwindow.h

```diff
@@ -205,6 +207,9 @@ private slots:
 #endif
     bool has_python;
     QMenu* toolbarMenu;
+    TrayIconBalloonShowReason baloonShowReason;
+    QString balloonClickDestinationPath;
```

Comment:
Why don't you store just a handle (or hash) to the torrent? Balloon clicks are rare operations, so it is not needed to optimize for speed, but here you increase memory consumption permanently.